src/cc/libbpf.c

```diff
@@ -762,47 +766,38 @@ static int bpf_attach_tracing_event(int progfd, const char *event_path,
     // Tracing events do not do CPU filtering in any cases.
     if (pid != -1)
       cpu = -1;
-    pfd = syscall(__NR_perf_event_open, &attr, pid, cpu, -1 /* group_fd */, PERF_FLAG_FD_CLOEXEC);
-    if (pfd < 0) {
+    *pfd = syscall(__NR_perf_event_open, &attr, pid, cpu, -1 /* group_fd */, PERF_FLAG_FD_CLOEXEC);
+    if (*pfd < 0) {
       fprintf(stderr, "perf_event_open(%s/id): %s\n", event_path, strerror(errno));
       return -1;
     }
   }
-  perf_reader_set_fd(reader, pfd);
 
-  if (perf_reader_mmap(reader, attr.type, attr.sample_type) < 0)
-    return -1;
-
-  if (ioctl(pfd, PERF_EVENT_IOC_SET_BPF, progfd) < 0) {
```

Comment:
Previously, if the `pfd` was created here (falling back to `debugfs`) and the`ioctl` fails, no would would clean them up. This commit fixes the issue and documented that caller should do the clean-up.